datahub-web-react/src/app/search/filters/useGetBrowseV2LabelOverride.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect } from 'react';
+import { useGetEntityDisplayNameLazyQuery } from '../../../graphql/search.generated';
+import { getLastBrowseEntryFromFilterValue } from './utils';
+import { BROWSE_PATH_V2_FILTER_NAME } from '../utils/constants';
+import EntityRegistry from '../../entity/EntityRegistry';
+
+export default function useGetBrowseV2LabelOverride(
+    filterField: string,
+    filterValue: string,
+    entityRegistry: EntityRegistry,
+) {
+    const [getEntityDisplayName, { data, loading }] = useGetEntityDisplayNameLazyQuery();
+
+    useEffect(() => {
+        if (filterField === BROWSE_PATH_V2_FILTER_NAME) {
+            const lastBrowseEntry = getLastBrowseEntryFromFilterValue(filterValue);
+            if (lastBrowseEntry.includes('urn:li:')) {
```

Comment:
tiny nit: can be extracted into an "isEntityRef" function or something of the like.

Reply:
nice